src/solvers/refinement/string_refinement_util.cpp

```diff
@@ -228,6 +228,24 @@ exprt interval_sparse_arrayt::at(std::size_t index) const
   return default_value;
 }
 
+array_exprt interval_sparse_arrayt::concretize(
+  std::size_t size,
+  const typet &index_type) const
+{
+  const array_typet array_type(
+    default_value.type(), from_integer(size, index_type));
+  array_exprt array(array_type);
+
+  std::size_t current_index = 0;
+  for(const auto &pair : entries)
+  {
+    for(; current_index <= pair.first && current_index < size; ++current_index)
+      array.operands().push_back(pair.second);
+  }
```

Comment:
I think you could remove `current_index` and add an if statement in this loop:

```c++
if (array.operands().size() < size)
  return array;
```

This should make this function a little bit more obvious.